contracts/v2/sovereignChains/GlobalExitRootManagerL2SovereignChain.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: AGPL-3.0
+
+pragma solidity 0.8.20;
+import "../../PolygonZkEVMGlobalExitRootL2.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+// WARNING: not audited
+
+/**
+ * Contract responsible for managing the exit roots for the Sovereign chains and global exit roots
+ */
+contract GlobalExitRootManagerL2SovereignChain is
+    PolygonZkEVMGlobalExitRootL2,
+    Initializable
+{
+    // globalExitRootUpdater address
+    address public globalExitRootUpdater;
+
+    // Inserted GER counter
+    uint256 public insertedGERCount;
+
+    /**
+     * @dev Emitted when a new global exit root is inserted
+     */
+    event InsertGlobalExitRoot(bytes32 indexed newGlobalExitRoot);
+
+    /**
+     * @dev Emitted when the last global exit root is removed
+     */
+    event RemoveLastGlobalExitRoot(bytes32 indexed removedGlobalExitRoot);
+
+    modifier onlyGlobalExitRootUpdater() {
```

Comment:
Following the rest of contrats convention, modifiers go after consturctor and initialization